Bulldozer.CSV/Maps/Business.cs

```diff
@@ -0,0 +1,832 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+//
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Linq;
+using Bulldozer.Model;
+using Bulldozer.Utility;
+using Rock;
+using Rock.Data;
+using Rock.Model;
+using Rock.Web.Cache;
+using static Bulldozer.CSV.CSVInstance;
+using static Bulldozer.Utility.Extensions;
+using static Bulldozer.Utility.CachedTypes;
+
+namespace Bulldozer.CSV
+{
+    /// <summary>
+    /// Partial of CSVComponent that holds the People import methods
+    /// </summary>
+    partial class CSVComponent
+    {
+        /// <summary>
+        /// Loads the Business data.
+        /// </summary>
+        private int ImportBusinesses()
+        {
+            ReportProgress( 0, "Preparing Business data for import..." );
+
+            // Let's deal with family groups first
+            var familiesToCreate = new List<Group>();
+            var rockContext = new RockContext();
+            var familyGroupTypeId = GroupTypeCache.GetFamilyGroupType().Id;
+            var importDateTime = RockDateTime.Now;
+
+            var families = this.BusinessCsvList.Where( b => !this.FamilyDict.ContainsKey( string.Format( "{0}^B_{1}", ImportInstanceFKPrefix, b.Id ) ) )
+                                                .GroupBy( b => b.Id )
+                                                .Select( a => new
+                                                {
+                                                    BusinessId = a.Key,
+                                                    Campus = a.Select( b => b.Campus ).FirstOrDefault(),
+                                                    CreatedDate = a.Select( b => b.CreatedDateTime ).FirstOrDefault(),
+                                                    ModifiedDate = a.Select( b => b.ModifiedDateTime ).FirstOrDefault(),
+                                                    BusinessName = a.Select( b => b.Name ).FirstOrDefault(),
+                                                } );
+
+            this.ReportProgress( 0, string.Format( "Creating {0} new Business Family Group records...", families.Count() ) );
+
+            foreach ( var family in families )
+            {
+                var newFamily = new Group
+                {
+                    GroupTypeId = familyGroupTypeId,
+                    Name = family.BusinessName,
+                    ForeignId = family.BusinessId.AsIntegerOrNull(),
+                    ForeignKey = string.Format( "{0}^B_{1}", ImportInstanceFKPrefix, family.BusinessId ),
+                    CreatedDateTime = family.CreatedDate.ToSQLSafeDate() ?? importDateTime,
+                    ModifiedDateTime = family.ModifiedDate.ToSQLSafeDate() ?? importDateTime
+                };
+
+                if ( family.Campus != null && family.Campus.CampusId.ToIntSafe( 0 ) > 0 )
```

Comment:
Validate the need of this logic. It appears the second condition of this outer if negates the need for the inner if.